### PR TITLE
[MM-402] Remove react-markdown package and use functions exposed by mattermost webapp

### PR DESCRIPTION
**Pull request by ayusht2810**
#### Summary
- Removed react-markdown package and used functions exposed by mattermost webapp 
- Converted `index.js` file to `index.ts` present in root of webapp folder

##### Before
image

##### After
image

#### Ticket Link
https://github.com/mattermost/mattermost-plugin-gitlab/security/dependabot/11

#### What to test?
- Complete test of link tooltip feature

#### Checklist
- [x] Completed dev testing
- [x] `make test` Ran test cases and ensured they are passing
- [x] `make check-style` Ran style check and ensured both webapp and server pass the checks

### CODE REVIEW

**Review by mickmister — APPROVED** [review]
LGTM, nice work @ayusht2810 :+1:

@ayusht2810 Just curious, are PRs like this being reviewed by the Brgithscout team as well? I see @raghavaggarwal2308 being requested for review on some of these

File: webapp/src/components/link_tooltip/index.tsx [hunk 1]
```diff
@@ -148,11 +148,7 @@ const LinkTooltip = ({href, show}: Props) => {
                             <span className='mr-number'>{`#${data.iid}`}</span>
                         </a>
                         <div className='markdown-text mt-1 mb-1'>
-                            <ReactMarkdown
-                                source={data.description}
-                                disallowedTypes={['heading']}
```
[the remark this answers is not in the document]
Comment (ayusht2810) [reply webapp/src/components/link_tooltip/index.tsx:153]: @mickmister With the new changes, `heading` tags will be shown something like this 
image

According to the documentation of [react-markdown](https://www.npmjs.com/package/react-markdown/v/3.0.1#options) the heading node should ignored, not sure why it was not happening.

File: webapp/src/components/link_tooltip/index.tsx [hunk 2]
```diff
@@ -148,11 +148,7 @@ const LinkTooltip = ({href, show}: Props) => {
...
-                                linkTarget='_blank'
-                            />
+                            {messageHtmlToComponent(formatText(data.description), false)}
```
[the remark this answers is not in the document]
Comment (ayusht2810) [reply webapp/src/components/link_tooltip/index.tsx:151]: Updated the PR description

**Comment by ayusht2810** [comment]
@mickmister We usually assign PRs created by us to one of our team member as well

File: webapp/src/index.ts [hunk 3]
```diff
@@ -97,7 +97,7 @@ class PluginClass {
     }
 
     deinitialize() {
-        document.removeEventListener('click', activityFunc);
+        document.removeEventListener('click', activityFunc as EventListenerOrEventListenerObject);
```
Comment (mickmister) [inline webapp/src/index.ts:100]: We could wrap this in an if statement instead of doing `as` here

**Review by mickmister — APPROVED** [review]

**Comment by ayusht2810** [comment]
@mickmister Should I move the PR for QA review?

**Review by AayushChaudhary0001 — APPROVED** [review]
The PR above has been tested for the following conditions:-

- Adding the text which is bold, italic and striked through
- Adding code snippets
- Adding links

The PR was working fine for all the above conditions, LGTM. Approved.

**Review by mickmister — APPROVED** [review]

**Pull request merged**